node/src/neighborhood/mod.rs

```diff
@@ -4319,6 +4869,8 @@ mod tests {
         let mut subject = neighborhood_from_nodes(&root_node, Some(&neighbor_node));
         let persistent_config = PersistentConfigurationMock::new();
         subject.persistent_config_opt = Some(Box::new(persistent_config));
+        assert!(subject.gossip_acceptor_opt.is_none());
+        assert!(subject.gossip_producer_opt.is_none());
```

Comment:
Probably one of these isn't necessary, unless there's some side-effect stuff going on in the background, in which case we need either comments (suboptimal) or redesign (preferred)!